test/subgraph.cpp

```diff
@@ -18,6 +18,27 @@
 
 // UNDER CONSTRUCTION
 
+
+// This is a helper function to recusively compare two subgraphs.
```

Comment:
Please add more detail about what it is testing, and the motivation for testing it.